_pages/documentation/gem5-stdlib/7-using-local-data-sources.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: documentation
+title: Using local Data sources in gem5
+parent: gem5-standard-library
+doc: gem5 documentation
+permalink: /documentation/gem5-stdlib/using-local-resources
+author: Harshil Patel
+---
+
+gem5 supports using local data sources in the form of a MongoDB Atlas and JSON datasource. To utilize data sources other than the main gem5 resources database, you will need to override the gem5-resources-config.
+
+MongoDB Config Format:
+```json
+{
+    "sources":{
+        "example-atlas-config": {
+            "dataSource": "datasource name",
+            "database": "database name",
+            "collection": "collection name",
+            "url": "Atlas data API URL",
+            "authUrl": "Atlas authentication URL",
+            "apiKey": "API key",
+            "isMongo": true
+        }       
+    }
+}
+```
+JSON Config Format:
+
+```json
+{
+    "sources":{
+        "example-json-config": {
+            "url": "local path to JSON file or URL to a JSON file",
+            "isMongo": false
+        }
+    }
+}
+```
+### Updating gem5 Resources Config:
+
+You can update the gem5 resources config in the following ways:
```

Comment:
I think we should rephrase L42, L44, L46, and L48. The problem is that it is unclear what `GEM5_CONFIG`, `GEM5_RESOURCE_JSON`, and `GEM5_RESOURCE_JSON_APPEND`. It is also unclear what happens when we modify more than one of the mentioned variables at the same time. Also, since there are many ways of modifying the config, is there a preferable way of doing this? I think we should have fewer ways of modifying the configs.